### PR TITLE
Proposed API for lists in tricorder.

### PR DESCRIPTION
**Pull request by keep94**
This is the proposed API for tricorder lists. This PR isn't meant to be merged. It is here only for feedback on the API changes.

### CODE REVIEW

File: go/tricorder/api.go [hunk 1]
```diff
@@ -250,6 +250,48 @@ func (d *NonCumulativeDistribution) Count() uint64 {
...
+//
+// aSliceMutable lets tricorder know whether or not caller plans to
+// modify aslice in the future. If caller passes ImmutableSlice or
+// false for aSliceMutable and later modifies aslice, the results
+// are undefined. If caller passes MutableSlice or true and later
+// modifies aslice, tricorder will continue to report the original
+// values in aslice.
+//
+// To change the values in a List, caller must use the Change method.
+func NewList(aslice interface{}, aSliceMutable bool) *List {
```
Comment (rgooch) [inline go/tricorder/api.go:282]: Perhaps use "aSlice" as the variable name? That seems more consistent with the common conventions.
Also, I think "sliceIsMutable" might be more clear. Or perhaps "sliceMutability".

**Comment by rgooch** [comment]
Overall, API LGTM.

**Comment by keep94** [comment]
So sorry, Just now saw the aslice -> aSlice comment. I will fix tomorrow.
That fix is not in the pull request I sent you today.

File: go/tricorder/api.go (hunk 1 above)
Comment (keep94) [inline go/tricorder/api.go:282]: Done.

**Pull request closed without merging**